res/controllers/Pioneer-DDJ-200-scripts.js

```diff
@@ -69,338 +73,438 @@ DDJ200.shutdown = function() {
-        eq = 3;
-    }
-    var deckNo = group.substring(24, 25);
-    // var deckNo = group.match("hannel.")[0].substring(6); // more general
-    // var deckNo = script.deckFromGroup(group); // working after fix
-    // https://github.com/mixxxdj/mixxx/pull/3178 only
-    var vDeckNo = DDJ200.vDeckNo[deckNo];
-    var vgroup = group.replace("Channel" + deckNo, "Channel" + vDeckNo);
-    engine.setValue(vgroup, "parameter" + eq, val);
-};
-
-DDJ200.super1 = function(channel, control, value, status, group) {
-    var val = script.absoluteNonLin(value, 0, 0.5, 1);
-    var deckNo = group.substring(26, 27);
-    //var deckNo = group.match("hannel.")[0].substring(6); // more general
-    //var deckNo = script.deckFromGroup(group); // working after fix
-    // https://github.com/mixxxdj/mixxx/pull/3178 only
-    var vDeckNo = DDJ200.vDeckNo[deckNo];
-    var vgroup = group.replace("Channel" + deckNo, "Channel" + vDeckNo);
-    engine.setValue(vgroup, "super1", val);
+        };
+    };
 };
 
-DDJ200.cueDefault = function(channel, control, value, status, group) {
-    if (value) { // only if button pressed, not releases, i.e. value === 0
-        var vDeckNo = DDJ200.vDeckNo[script.deckFromGroup(group)];
-        var vgroup = "[Channel" + vDeckNo + "]";
-        if (!DDJ200.vDeck[vDeckNo]["jogEnabled"]) {  // if jog top is touched
-            engine.setValue(vgroup, "cue_set", true);
+DDJ200.padLEDsOff = function() {                         // turn off LED buttons:
+    for (let i = 0; i <= 1; i++) {
+        midi.sendShortMsg(0x96 + i, 0x63, 0x00);      // set headphone master
+        midi.sendShortMsg(0x90 + i, 0x54, 0x00);      // pfl headphone
+        midi.sendShortMsg(0x90 + i, 0x58, 0x00);      // beat sync
+        midi.sendShortMsg(0x90 + i, 0x0B, 0x00);      // play
+        midi.sendShortMsg(0x90 + i, 0x0C, 0x00);      // cue
+        for (let j = 0; j <= 8; j++) {
+            midi.sendShortMsg(0x97 + 2 * i, j, 0x00); // hotcue
+        };
+    };
+};
+
+DDJ200.padModes = ["Hotcue", "Loop", "Effect", "Jump"];
+DDJ200.padModeIndex = 0;
+DDJ200.transFxButton = new components.Button({
+    blinkConnection: undefined,
+    input: function(_channel, _control, value, _status, _g) {
+        if (value) { // only if button pressed
+            if (DDJ200.padModeIndex === (DDJ200.padModes.length - 1)) {
+                DDJ200.padModeIndex = 0;
+            } else {
+                if (DDJ200.leftDeck.shiftPressed || DDJ200.rightDeck.shiftPressed) {
+                    DDJ200.padModeIndex = DDJ200.padModes.length - 1;
+                } else {
+                    DDJ200.padModeIndex = (DDJ200.padModeIndex + 1) % (DDJ200.padModes.length - 1);
+                };
+            };
+            DDJ200.leftDeck.pads.forEach(function(e) {
+                e.updateOutKey();
+            });
+            DDJ200.rightDeck.pads.forEach(function(e) {
+                e.updateOutKey();
+            });
+            this.output();
+        };
```

Comment:
rework with `ComponentContainter`s (see previous comments). Also consider using `script.posMod` and/or a helper like `NS6II.RinBufferView` (feel free to copy-paste).